oauth2/src/main/java/com/okta/spring/oauth/OktaOAuthProperties.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2017 Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.spring.oauth;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ConfigurationProperties("okta")
+public class OktaOAuthProperties {
+
+
+    private ClientProperties client = new ClientProperties();
+
+    private OAuthProperties oauth2 = new OAuthProperties();
+
+    private Map<String, Object> extraWidgetConfig = new HashMap<>();
+
+
+    @PostConstruct
+    public void init() {
+        // make sure 'features' is a map
+        Object[] idps = convertToArray(extraWidgetConfig.get("idps"));
+        if (idps.length > 0) {
+            extraWidgetConfig.put("idps", idps);
+        }
+    }
+
+    private Object[] convertToArray(Object array) {
+        if (array instanceof Map) {
+            Map map = (Map) array;
+            return map.values().toArray();
+        }
+        return new Object[0];
+    }
+
+    public ClientProperties getClient() {
+        return client;
+    }
+
+    public void setClient(ClientProperties client) {
+        this.client = client;
+    }
+
+
+    public OAuthProperties getOauth2() {
```

Comment:
seems like this should be: `getOAuthProperties()` or `getOAuth2Properties` (which also means that the variable should probs be renamed too)

Reply:
See above, that might help clear up the confusion (instead of forcing them all in a single class)?